src/chromium/crNetworkManager.ts

```diff
@@ -91,17 +97,19 @@ export class CRNetworkManager {
   }
 
   async setCacheEnabled(enabled: boolean) {
-    this._userCacheDisabled = !enabled;
     await this._updateProtocolCacheDisabled();
   }
 
   async setRequestInterception(value: boolean) {
-    this._userRequestInterceptionEnabled = value;
     await this._updateProtocolRequestInterception();
   }
 
-  async _updateProtocolRequestInterception() {
-    const enabled = this._userRequestInterceptionEnabled || !!this._credentials;
+  private _userRequestInterceptionEnabled() : boolean {
+    return !!this._page.browserContext()._options.interceptNetwork;
+  }
+
+  private async _updateProtocolRequestInterception() {
```

Comment:
This stuff is too complicated 😞

Reply:
it is, let me change that in a follow-up.